### ISSUE TITLE
Using pogo-optimizer is modifying in-game data

### ISSUE THREAD

**Issue by lennykano** [justinleewells/pogo-optimizer#124]
When I have the proxy switched on, the game shows me as having 0 pokemon in gyms. The ones I know are currently in gyms appear in my inventory with full health. As soon as I turn off the Wi-Fi connection, it goes back to the correct number. This concerns me, as the only reason I was comfortable using this was due to the assertion that none of the data is modified, and would therefore be transparent to Niantic.

**Comment by justinleewells** [comment]
It would not be transparent to Niantic. It's only modifying the data that's sent to you, not sent to them. I am curious why it's modifying that data at all, however.

**Comment by lennykano** [comment]
> It would not be transparent to Niantic. It's only modifying the data that's sent to you, not sent to them.

I'm confused, those two statements seem to contradict each other.

But yeah, I assumed it was just another bug with the game, but it definitely only happens when I'm using the proxy setup for this. Have tested multiple times to confirm, and happened at home and a friends place, only on the proxied connection.

**Comment by justinleewells** [comment]
The data sent to you and the data sent to the server are two different things. I believe you that it's being modified, I just think that it's an issue with the POGOProtobuf library.

**Comment by lennykano** [comment]
I didn't think anything at all was modified, just intercepted to read the data from and passed along as is, which obviously doesn't appear to be the case. Not sure why it would even need to be modified? Slightly less comfortable now that I know that what the app ends up seeing is not the same as what the server sent down.

At any rate, sounds like it's just a flag being dropped. Not a big deal since I can't reach any gyms from anywhere I'm going to be connected over the proxy, just thought it worth mentioning.

**Comment by mathiasbynens** [comment]
@lennykano Would you mind filing this as a POGOProtobuf bug as per https://github.com/justinleewells/pogo-optimizer/issues/124#issuecomment-235451351?

**Comment by lennykano** [comment]
Sure, is https://github.com/AeonLucid/POGOProtos the project you're referring to? Google isn't helping much when I search for POGOProtobuf.

**Comment by mathiasbynens** [comment]
Hmm, I guess it would be https://github.com/rastapasta/pokemon-go-protobuf-node since that’s what https://github.com/rastapasta/pokemon-go-mitm-node (which pogo-optimizer depends on) uses. But @justinleewells would know for sure.

**Comment by mathiasbynens** [comment]
cc @rastapasta

**Comment by lennykano** [comment]
No probs, I'm gonna hold off making an issue just yet because I just got home, ran a `git pull` and started it up to check my IVs for todays catch, and the bug isn't happening now. Don't have time to investigate now but I'll check the other machine/network tomorrow morning and see if the pulling the latest code fixes it there too. Synced it this morning (13-14 hours ago) before the bug happened, so if a change has fixed it it's something pretty recent.

**Comment by lennykano** [comment]
OK just tested at my friends place and it's still happening on the latest version. If I use the proxied connection, the game thinks I have 0 defenders in gyms, as soon as I switch over to the non-proxied one it says the correct number. Not sure why it didn't happen at home last night.

It looks like the repo I linked is used by https://github.com/rastapasta/pokemon-go-protobuf-node, and I found that the issue has already been reported there: https://github.com/AeonLucid/POGOProtos/issues/43
